src/data-structures/arrays/numbers/arraysOfNums.js

```diff
@@ -11,16 +11,15 @@ export const generateArrayOfNums = (arrayCount, itemCount, range) => {
     return arr;
 }
 
-export const checkArrayOfNums = (dataSet) => {
-    if (!Array.isArray(dataSet)) {
-        return false;
-    }
+const checkArrayOfNums = (dataSet) => {
+    if (!Array.isArray(dataSet)) return false;
+    if (dataSet.length === 0) return true;
 
     for (let i = 0; i < dataSet.length; i++) {
-        if (typeof dataSet[i] !== 'number' || Number.isNaN(dataSet[i]) || !Number.isFinite(dataSet[i])) {
-            return false;
-        }
-
+        if (typeof dataSet[i] !== 'number' || Number.isNaN(dataSet[i]) || !Number.isFinite(dataSet[i])) return false;
         return true;
```

Comment:
can't be correct. After checking the first index, like dataset[0] inside the loop it does return right away true or false, without going to the second item